### DISCUSSION TITLE
Search Page Integration

### DISCUSSION THREAD

**Discussion by panchi64** [uprm-inso4101-2023-2024-S1/semester-project-tutoring-app#63] [General]
Alrighty drop the convo here team 2.
@Jonathan-Rod @williammartinez10 @gustavosotero @RicardoJRSs @Yahid1 @yahirfigueroa y everyone else I can't remember rn

**Comment by Yahid1** [comment]
Can someone explain how the search bar component works?? Cause its giving the wrong results but maybe im using it wrong.

**Reply by Yahid1** [reply]
The problem i find is that it doesnt filter by combination of letters it filters by searching if the name has all the letters inside it. 
Example:  I put = ae 
it returns a list of all the names that have ae:  Gerardo , Alice , Jaime


Instead of the combination in sequence
Example : Laeria , Jaenia  or Ricardae

**Reply by Jonathan-Rod** [reply]
**My latest commit:**
- In summary, this `MySearchBar` component provides a basic search interface, where the user can enter search text and view the results in a list of cards. 
- The actual search logic should be implemented in the part marked as `// TODO` within the `handleSearch` function.

![image](https://github.com/uprm-inso4101-2023-2024-S1/semester-project-tutoring-app/assets/60677254/003ba3dd-2359-47bc-baed-611a8aa8e13a)

**Reply by Yahid1** [reply]
![image](https://github.com/uprm-inso4101-2023-2024-S1/semester-project-tutoring-app/assets/94016523/e62c84d2-21a2-4d09-a421-fa63c64b9580)

Do you think you can fix what i described earlier based on this code?

**Reply by Jonathan-Rod** [reply]
Let me think about it.

**Reply by Yahid1** [reply]
![image](https://github.com/uprm-inso4101-2023-2024-S1/semester-project-tutoring-app/assets/94016523/b17cb443-8f96-4d57-81fc-1a46fef780c5)

This is another example

**Reply by Jonathan-Rod** [reply]
**Let me propose:**

1. **Importing the List of Tutors:**
Importing the list of tutors within the component file instead of passing it as props can be a valid approach if the list is defined in another module.

2. **Filtering Approach:**
Modifying the filtering approach to filter tutors whose names contain the search text (partial matching) is more appropriate if you want to find tutors whose names have any part of the search text, not just an exact match.

3. **Removing `searchText !== ""`:**
You can remove the condition `searchText !== ""` from the rendering logic if you want to simplify the code and rely on the fact that `searchResults` will only contain matching tutors when there's some search text.

**Comment by Jonathan-Rod** [comment]
I need to know all the requirements we need to perform a search.

Please comment below your expectations about:

1. **Filtering**
2. **Rendering**

**Reply by panchi64** [reply]
In an ideal world... I'd expect as a user to be able to search by department, class, and tutor. 

However, If a user can just manage to search for a class and it can show me the tutors available for said class that works out just fine.

**Reply by panchi64** [reply]
I misread this completely 🙃

**Reply by Jonathan-Rod** [reply]
Well, a new feature as default options under the search bar as buttons that when clicked, set the list of results with a certain order, or from a certain department.


**But focus on the actual content of the search bar**, to find out what changes a stable version.

**Comment by Yahid1** [comment]
In the  serv_page branch I finished an implementation of the search page. I connected all the components created using the data extrated from the listDepartments.js file that returns a List with all the data orginized.

Here is how the data looks like:
![image](https://github.com/uprm-inso4101-2023-2024-S1/semester-project-tutoring-app/assets/94016523/573e534d-2e28-4210-a625-9dd6f5f33a02)


This is how it looks on the App in a browser:
![](https://cdn.discordapp.com/attachments/1146484069660639233/1153728330735030283/image.png)


I will also mention there is a new bug I found. When you filter, it keeps the wrong course expanded.

Here is an example video taken in the IOS app version showing the bug:

https://github.com/uprm-inso4101-2023-2024-S1/semester-project-tutoring-app/assets/94016523/2ccd2db7-d9b9-489f-9f95-fa4adca3c284

It is a bug because It’s not the course that I expanded originally.

To replicate the bug you have to:

1) Search INGE 
2) Expand tutors on the Inge department
3) Then delete the search
And you’ll find CIIC tutors expanded

Which shouldn’t happen cause I only expanded under INGE.

**Reply by Yahid1** [reply]
The correct behavior should be that it keeps the tutors from the INGE department expanded and only when they get filtered out should they get reset into not expanded.

**Reply by Yahid1** [reply]
I will create an Issue for this bug. For anyone to fix it.
